warehouse-management-service/pkg/database/postgres/shelf_block_test.go

```diff
@@ -0,0 +1,679 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"github.com/golang/mock/gomock"
+	_ "github.com/lib/pq"
+	"testing"
+	wms "warehouse-management-service"
+)
+
+func TestGetShelfBlockByIdTx(t *testing.T) {
+	warehouse := wms.Warehouse{
+		Id:        "85bd3b85-ad4d-4224-b589-fb2a80a6ce45",
+		Name:      "test_get_shelf_block_by_id",
+		Latitude:  12.9716,
+		Longitude: 77.5946,
+	}
+
+	shelfBlock := wms.ShelfBlock{
+		Id:          "get_test",
+		Aisle:       "1",
+		Rack:        "1",
+		StorageType: "regular",
+		WarehouseId: "85bd3b85-ad4d-4224-b589-fb2a80a6ce45",
+	}
+
+	tx, err := shelfService.db.Begin()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer tx.Rollback()
+
+	// insert warehouse and shelf_block records
+	warehouseQuery := "INSERT INTO warehouse (id, name, geolocation) VALUES ($1, $2, point($3, $4))"
+	shelfBlockQuery := "INSERT INTO shelf_block(id, aisle, rack, storage_type, warehouse_id) VALUES ($1, $2, $3, $4, $5)"
+
+	_, err = tx.ExecContext(
+		context.Background(),
+		warehouseQuery,
+		warehouse.Id,
+		warehouse.Name,
+		warehouse.Longitude,
+		warehouse.Latitude,
+	)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	_, err = tx.ExecContext(
+		context.Background(),
+		shelfBlockQuery,
+		shelfBlock.Id,
+		shelfBlock.Aisle,
+		shelfBlock.Rack,
+		shelfBlock.StorageType,
+		shelfBlock.WarehouseId,
+	)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	shelfBlockFromDB, err := shelfService.queries.getShelfBlockByIdTx(
+		context.Background(),
+		tx,
+		shelfBlock.Id,
+	)
+
+	if shelfBlockFromDB != shelfBlock {
+		t.Errorf("expected: %v, got: %v", shelfBlock, shelfBlockFromDB)
+	}
+}
+
+func TestGetShelfBlockByIdTxError(t *testing.T) {
+	tx, err := shelfService.db.Begin()
+	if err != nil {
+		t.Error(err)
+	}
+	defer tx.Rollback()
+
+	_, err = shelfService.queries.getShelfBlockByIdTx(context.Background(), tx, "bad_id")
```

Comment:
Should we assert on the actual error also and not just checking if it's nil or not? What do you think?

Reply:
The intent here is to check if `getShelfBlockByIdTx` returns an error when the underlying query returns an error. This is why I considered validating the error itself to be unnecessary (not intended to test the library).

One thing I can think of why checking for the exact error makes sense is to ensure that the error is returned as is without any abstraction. This is what is intended here and if we want to do so in the future, the tests will have to be updated as well making the robust.

Reply:
Changed this to assert on the actual error